qutebrowser/mainwindow/statusbar/bar.py

```diff
@@ -218,15 +221,17 @@ def _on_config_changed(self, option):
         elif option == 'statusbar.widgets':
             self._draw_widgets()
 
-    def _draw_widgets(self):
+    def _draw_widgets(self):  # noqa: C901
```

Comment:
I tried to simplify this function, but couldn't find a way that didn't either make it more complicated in the eyes of flake8 or explode the code into ten different functions.